cmd/terramate/cli/project.go

```diff
@@ -225,25 +225,38 @@ func (p project) checkDefaultRemote() error {
 	)
 }
 
-func (p *project) checkLocalDefaultIsUpdated() error {
+func (p *project) checkRemoteDefaultBranchIsReachable() error {
 	gitcfg := p.gitcfg()
+
+	remoteDesc := fmt.Sprintf("remote(%s/%s)", gitcfg.DefaultRemote, gitcfg.DefaultBranch)
+
+	logger := log.With().
+		Str("HEAD", p.headCommit()).
+		Str("default_branch", remoteDesc).
+		Str("default_hash", p.remoteDefaultCommit()).
+		Logger()
+
+	outOfDateErr := errors.E(
+		ErrCurrentHeadIsOutOfDate,
+		"Please merge the latest changes from %s into this branch",
+		remoteDesc,
+	)
+
 	mergeBaseCommitID, err := p.git.wrapper.MergeBase(p.headCommit(), p.remoteDefaultCommit())
 	if err != nil {
-		return fmt.Errorf(
-			"the reference %s/%s is not reachable from HEAD: %w",
-			gitcfg.DefaultRemote,
-			gitcfg.DefaultBranch,
-			err,
-		)
+		logger.Debug().Msg(err.Error())
+		return outOfDateErr
 	}
 
+	logger = logger.With().
+		Str("merge_base", mergeBaseCommitID).
+		Logger()
+
 	if mergeBaseCommitID != p.remoteDefaultCommit() {
-		return errors.E(
-			ErrOutdatedLocalRev,
-			"remote %s/%s != HEAD",
-			gitcfg.DefaultRemote,
-			gitcfg.DefaultBranch,
+		logger.Debug().Msgf(
+			"The %s is not the merge-base of current HEAD", remoteDesc,
 		)
```

Comment:
remoteDesc is already in the context of current logger

```suggestion
		logger.Debug().
		  Str("merge_base_hash", mergeBaseCommitID).
		  Msg("The default branch is not equal to the common merge-base of HEAD")
```